drwmutex.go

```diff
@@ -89,15 +96,22 @@ func (dm *DRWMutex) Lock() {
 func (dm *DRWMutex) RLock() {
 
 	isReadLock := true
-	dm.lockBlocking(isReadLock)
+	dm.lockBlocking(time.Duration(1<<63-1), isReadLock)
+}
+
+// GetRLock tries to get a read lock on dm before the timeout occurs.
+func (dm *DRWMutex) GetRLock(timeout time.Duration) (locked bool) {
+
+	isReadLock := true
+	return dm.lockBlocking(timeout, isReadLock)
 }
 
 // lockBlocking will acquire either a read or a write lock
 //
 // The call will block until the lock is granted using a built-in
 // timing randomized back-off algorithm to try again until successful
-func (dm *DRWMutex) lockBlocking(isReadLock bool) {
-	doneCh := make(chan struct{})
+func (dm *DRWMutex) lockBlocking(timeout time.Duration, isReadLock bool) bool {
```

Comment:
Unfortunately I am not able to come up with better name for `timeout`. 
The essence of `timeout` behavior is below;

* If all `NetLocker` are locked without given timeout, the lock is returned
* If quorum number of `NetLocker` are locked within given timeout, the lock is returned
* If quorum number of `NetLocker` are not locked within given timeout, retry remaining `NetLocker`.
* If quorum number of `NetLocker` are failed within given timeout, return failure.

In all the cases, timeout is honored after first attempt.  Not sure which name is meaningful.  Since `dsync` is a general purpose distributed lock system, I was thinking of better naming for timeout.

Irrespective of the naming, we would need to have proper godoc explain what the timeout means for `GetLock()`.  Below description doesn't convey the indented meaning.

```
 GetLock tries to get a write lock on dm before the timeout occurs.
```